pkg/deploy/service_accounts.go

```diff
@@ -49,43 +56,26 @@ func (s *ServiceAccounts) fetchServiceAccount(nsName string, saName string) (str
 		return "", fmt.Errorf("Internal inconsistency: Expected service account name to not be empty")
 	}
 
-	sa, err := s.coreClient.CoreV1().ServiceAccounts(nsName).Get(context.Background(), saName, metav1.GetOptions{})
+	tokenMgr := token.NewManager(s.coreClient, s.log)
+
+	t, err := tokenMgr.GetServiceAccountToken(nsName, saName, &authenticationv1.TokenRequest{
+		Spec: authenticationv1.TokenRequestSpec{
+			ExpirationSeconds: getInt64Point(5000), // 5 minutes
+		},
+	})
 	if err != nil {
-		return "", fmt.Errorf("Getting service account: %s", err)
+		return "", fmt.Errorf("Failed to get service account token: %s", err)
```

Comment:
nit: dont commonly include "Failed" in the error itself (it's an error, so it did indeed fail)